core/src/main/java/org/infinispan/interceptors/distribution/L1NonTxInterceptor.java

```diff
@@ -174,27 +180,35 @@ private Object performL1Lookup(InvocationContext ctx, VisitableCommand command,
          if (log.isTraceEnabled()) {
             log.tracef("Found current request for key %s, waiting for their invocation's response", key);
          }
-         Object returnValue;
-         try {
-            returnValue = presentSync.get(replicationTimeout, TimeUnit.MILLISECONDS);
-         } catch (TimeoutException e) {
-            // This should never be required since the status is always set in a try catch above - but IBM
-            // doesn't...
-            log.warnf("Synchronizer didn't return in %s milliseconds - running command normally!",
-                  replicationTimeout);
-            // Always run next interceptor if a timeout occurs
-            return invokeNext(ctx, command);
-         } catch (ExecutionException e) {
-            throw e.getCause();
-         }
+         CompletionStage<Object> stage = blockingManager.supplyBlocking(() -> {
```

Comment:
I don't think it is worth it, but relooking at the code the TimeoutException path is no longer correct as it will run the command twice. We would need to optionally invoke next if there was a timeout in the if below. I will try to fix this tomorrow.